src/bootstrap/compile.rs

```diff
@@ -1161,9 +1161,16 @@ impl Step for Assemble {
             let dst_exe = exe("rust-llvm-dwp", target_compiler.host);
             let llvm_config_bin = builder.ensure(native::Llvm { target: target_compiler.host });
             if !builder.config.dry_run {
-                let llvm_bin_dir = output(Command::new(llvm_config_bin).arg("--bindir"));
-                let llvm_bin_dir = Path::new(llvm_bin_dir.trim());
-                builder.copy(&llvm_bin_dir.join(&src_exe), &libdir_bin.join(&dst_exe));
+                if target_compiler.host == build_compiler.host {
+                    let llvm_bin_dir = output(Command::new(llvm_config_bin).arg("--bindir"));
+                    let llvm_bin_dir = Path::new(llvm_bin_dir.trim());
+                    builder.copy(&llvm_bin_dir.join(&src_exe), &libdir_bin.join(&dst_exe));
+                } else {
+                    // if cross-compiling don't copy the llvm directory for the target instead
+                    let llvm_bin_dir = format!("build/{}/llvm/build/bin", target_compiler.host);
+                    let llvm_bin_dir = Path::new(&llvm_bin_dir);
+                    builder.copy(&llvm_bin_dir.join(&src_exe), &libdir_bin.join(&dst_exe));
```

Comment:
nit: you can move this very last line out of the conditional.

Reply:
good point, done.